### PR TITLE
Fix flaky test TestContainerCMD

### LINKED ISSUE #3984: [Flaky Test] TestContainerCMD
## Flaky Test

* Test Name: **TestContainerCMD**.
* Link: /home/ubuntu/agent/testing/integration/container_cmd_test.go:113.
* Branch: main/all.
* Artifact Link: [TEST-go-integration.out.json](https://github.com/elastic/elastic-agent/files/13811982/TEST-go-integration.out.json)

* Notes: The test uses the `container` subcomand outside a container. I believe that as the agent isn't built to run on a container, it's built with `github.com/elastic/elastic-agent/internal/pkg/release.allowUpgra=true`, making the agent upgradable. What in turn makes it try to start the watcher, but as it's neither running installed nor any cli flag is passed to adjust the Top path, the agent tries to start the watcher from the wrong location:
```
-> "failed to invoke rollback watcher: failed to start Upgrade Watcher: fork/exec /usr/share/elastic-agent/state/data/elastic-agent: no such file or directory"

{"log.level":"error","@timestamp":"2024-01-02T15:16:00.811Z","log.origin":{"file.name":"cmd/run.go","file.line":231},"message":"failed to invoke rollback watcher: failed to start Upgrade Watcher: fork/exec /usr/share/elastic-agent/state/data/elastic-agent: no such file or directory","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
```

### Stack Trace

```
    fixture.go:615: >> running binary with: [/tmp/TestContainerCMD2837925629/001/elastic-agent-8.12.0-SNAPSHOT-linux-x86_64/elastic-agent status --output json]
    container_cmd_test.go:113: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/container_cmd_test.go:113
        	Error:      	Condition never satisfied
        	Test:       	TestContainerCMD
        	Messages:   	Elastic-Agent did not report healthy. Agent status error: "", Agent logs
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:15:59.339Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":496},"message":"Starting enrollment to URL: https://f797766cb61e4903bc854e52cdd78684.fleet.us-west2.gcp.elastic-cloud.com:443/","ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.513Z","log.origin":{"file.name":"cmd/enroll_cmd.go","file.line":289},"message":"Elastic Agent has been enrolled; start Elastic Agent","ecs.version":"1.6.0"}
        	            	Successfully enrolled the Elastic Agent.
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.558Z","log.origin":{"file.name":"cmd/run.go","file.line":169},"message":"Elastic Agent started","log":{"source":"elastic-agent"},"process.pid":22908,"agent.version":"8.12.0","ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.810Z","log.origin":{"file.name":"upgrade/rollback.go","file.line":126},"message":"Starting upgrade watcher","log":{"source":"elastic-agent"},"path":"/usr/share/elastic-agent/state/data/elastic-agent","args":["/usr/share/elastic-agent/state/data/elastic-agent","watch","--path.config","/usr/share/elastic-agent/state","--path.home","/usr/share/elastic-agent/state/data"],"env":[],"dir":"","ecs.version":"1.6.0"}
        	            	{"log.level":"error","@timestamp":"2024-01-02T15:16:00.811Z","log.origin":{"file.name":"cmd/run.go","file.line":231},"message":"failed to invoke rollback watcher: failed to start Upgrade Watcher: fork/exec /usr/share/elastic-agent/state/data/elastic-agent: no such file or directory","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.811Z","log.origin":{"file.name":"cmd/run.go","file.line":252},"message":"APM instrumentation disabled","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.812Z","log.origin":{"file.name":"application/application.go","file.line":62},"message":"Gathered system information","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.883Z","log.origin":{"file.name":"application/application.go","file.line":68},"message":"Detected available inputs and outputs","log":{"source":"elastic-agent"},"inputs":["syncgateway/metrics","activemq/metrics","kafka/metrics","cloudbeat/cis_k8s","cloud_defend/control","log","enterprisesearch/metrics","haproxy/metrics","memcached/metrics","audit/auditd","audit/file_integrity","elasticsearch/metrics","kafka","mysql/metrics","endpoint","azure-blob-storage","cel","netflow","syslog","rabbitmq/metrics","cloudbeat/cis_eks","cloudbeat/vuln_mgmt_aws","synthetics/tcp","lumberjack","nats/metrics","audit/system","packet","http_endpoint","cometd","pf-elastic-symbolizer","pf-host-agent","windows/metrics","nginx/metrics","vsphere/metrics","synthetics/icmp","aws-cloudwatch","winlog","mqtt","logstash/metrics","uwsgi/metrics","azure/metrics","etcd/metrics","cloudbeat/cis_gcp","container","docker","postgresql/metrics","apache/metrics","prometheus/metrics","cloudbeat/cis_azure","redis","udp","filestream","redis/metrics","system/metrics","containerd/metrics","gcp/metrics","stan/metrics","synthetics/http","httpjson","http/metrics","pf-elastic-collector","synthetics/browser","cloudfoundry","entity-analytics","o365audit","unix","docker/metrics","kubernetes/metrics","cloudfoundry/metrics","statsd/metrics","zookeeper/metrics","fleet-server","awsfargate/metrics","apm","gcp-pubsub","gcs","beat/metrics","mssql/metrics","traefik/metrics","cloudbeat/cis_aws","aws-s3","azure-eventhub","linux/metrics","aws/metrics","sql/metrics","journald","tcp","kibana/metrics","mongodb/metrics","oracle/metrics","iis/metrics","jolokia/metrics","osquery","cloudbeat"],"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.883Z","log.origin":{"file.name":"capabilities/capabilities.go","file.line":48},"message":"Capabilities file not found in /usr/share/elastic-agent/state/capabilities.yml","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.883Z","log.origin":{"file.name":"application/application.go","file.line":74},"message":"Determined allowed capabilities","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:00.883Z","log.origin":{"file.name":"application/application.go","file.line":89},"message":"Loading baseline config from /usr/share/elastic-agent/state/elastic-agent.yml","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:01.172Z","log.origin":{"file.name":"application/application.go","file.line":160},"message":"Parsed configuration and determined agent is managed by Fleet","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:01.175Z","log.origin":{"file.name":"runtime/manager.go","file.line":209},"message":"Starting grpc control protocol listener on port 6789 with max_message_size 104857600","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:01.177Z","log.logger":"composable.providers.docker","log.origin":{"file.name":"docker/docker.go","file.line":44},"message":"Docker provider skipped, unable to connect: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
        	            	{"log.level":"info","@timestamp":"2024-01-02T15:16:01.842Z","log.origin":{"file.name":"fleet/fleet_gateway.go","file.line":148},"message":"Fleet gateway started","log":{"source":"elastic-agent"},"ecs.version":"1.6.0"}
    container_cmd_test.go:90: >> cleaning up: killing the Elastic-Agent process
```

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Fix flaky test TestContainerCMD by setting `STATE_PATH` when launching the agent using the `container` subcommand so the upgrade watcher can be launched from the right location.

## Why is it important?

The test was broken, blocking all PRs, then it was skipped. This brings back the test with a fix for it.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## How to test this PR locally

 - run the integration tet TestContainerCMD, it should pass. 

## Related issues

- Closes #3984 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by cmacknz — APPROVED** [review]

**Review by blakerouse — COMMENTED** [review]
What is the reason for having an integration test that is testing the Elastic Agent in a non-real world situation to begin with?

**Comment by cmacknz** [comment]
> What is the reason for having an integration test that is testing the Elastic Agent in a non-real world situation to begin with?

We made a change that broken the `container` command and otherwise have no test coverage for it. It would be better to actually run this in a container, but this is a nice stop gap until we update the integration test framework to allow it to provision machines running Docker containers produced by the CI build.

File: testing/integration/container_cmd_test.go [hunk 1]
```diff
@@ -105,6 +105,11 @@ func TestContainerCMD(t *testing.T) {
 		"FLEET_ENROLL=1",
 		"FLEET_URL="+fleetURL,
 		"FLEET_ENROLLMENT_TOKEN="+enrollmentToken.APIKey,
+		// As the agent isn't built for a container, it's upgradable, triggering
```
Comment (cmacknz) [inline testing/integration/container_cmd_test.go:108]: This test is still skipped so the green build is misleading right now.

**Review by cmacknz — CHANGES_REQUESTED** [review]
Requesting changes to have the test no longer be skipped.

**Pull request merged**